src/Main.cpp

```diff
@@ -75,26 +101,60 @@ class HostUartCommTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
   private:
     void task() override {
         IHostUart& hostUart = BSPContainer::getHostUart();
+        uint8_t buffer[5];
+        buffer[sizeof(buffer) - 1] = 0;
+
         while (true) {
             hostUart.send((const uint8_t*)"HELLO WORLD", sizeof("HELLO WORLD"));
             vTaskDelay(1000);
         }
     }
 };
 
+class TCPReadDemoTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
+  public:
+    TCPReadDemoTask(const char* taskName, UBaseType_t priority) :
+        AbstractTask(taskName, priority), m_logger(LoggerContainer::getLogger()) {}
+
+    ~TCPReadDemoTask() override = default;
+
+  private:
+    ILogger& m_logger;
+    void task() override {
+
+        uint8_t buffer[5];
+        buffer[sizeof(buffer) - 1] = 0;
+
+        // Wait for connection
+        while (true) {
+            if (std::optional<TCPClientWrapper> socket = SocketContainer::getHostClientSocket()) {
+                while (true) {
+                    auto ret = socket.value().receive(buffer, sizeof(buffer) - 1);
+                    (void)ret;
+                    m_logger.log(LogLevel::Info, "TCP RX: %s", buffer);
+                }
+            }
+            // Retry connection every 2s
+            vTaskDelay(2000);
+        }
+    }
+};
+
 class HostTCPCommTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
   public:
-    HostTCPCommTask(const char* taskName, UBaseType_t priority) :
-        AbstractTask(taskName, priority) {}
+    HostTCPCommTask(const char* taskName, UBaseType_t priority, TCPReadDemoTask& tcpReadTask) :
+        AbstractTask(taskName, priority), m_tcpReadDemoTask(tcpReadTask) {}
 
     ~HostTCPCommTask() override = default;
 
   private:
+    TCPReadDemoTask& m_tcpReadDemoTask;
```

Comment:
So the HostTCPCommTask is responsible for the tcpReadDemoTask? could it create it on construction instead of injecting it if it's responsible of it? IDK if it would make more sense

Reply:
All of this is gonna be removed the moment we have a dispatcher. Does it really give anything to make it clean now?